### PR TITLE
Lhotse dataloading: RIR augmentation and nemo/tarred input support for RIR and noise aug

### PR DESCRIPTION
**Pull request by pzelasko**
# What does this PR do ?

Extensions in Lhotse dataloading augmentations:
* RIR augmentation support (currently requires lhotse RecordingSet input, see e.g. https://github.com/lhotse-speech/lhotse/blob/master/lhotse/recipes/rir_noise.py )
* Broader input type support for noise augmentation. 
  * When a path is provided, it auto-detects
    * YAML input spec (.yaml)
    * lhotse non-tarred CutSet (.jsonl/.jsonl.gz)
    * NeMo non-tarred manifest (any other extension)
  * Also possible to provide a dict with the same keys as used for specifying the main training data, e.g.:

```yaml
noise_path:
  manifest_filepath: my_noises/manifest__OP_0..4_CL_.json
  tarred_audio_filepaths: my_noises/audio__OP_0..4_CL_.tar
```

**Collection**: ASR

# Changelog 
- Add specific line by line info of high level changes in this PR.

# Usage
* You can potentially add a usage example below

```python
# Add a code snippet demonstrating how to use this 
```

# GitHub Actions CI

The Jenkins CI system has been replaced by GitHub Actions self-hosted runners.

The GitHub Actions CI will run automatically when the "Run CICD" label is added to the PR.
To re-run CI remove and add the label again.
To run CI on an untrusted fork, a NeMo user with write access must first click "Approve and run".

# Before your PR is "Ready for review"
**Pre checks**:
- [ ] Make sure you read and followed [Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md)
- [ ] Did you write any new necessary tests?
- [ ] Did you add or update any necessary documentation?
- [ ] Does the PR affect components that are optional to install? (Ex: Numba, Pynini, Apex etc)
  - [ ] Reviewer: Does the PR have correct import guards for all optional libraries?
  
**PR Type**:
- [x] New Feature
- [ ] Bugfix
- [ ] Documentation

If you haven't finished some of the above items you can still open "Draft" PR.

## Who can review?

Anyone in the community is free to review the PR once the checks have passed. 
[Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md) contains specific people who can review PRs to various areas.

# Additional Information
* Related to # (issue)

### CODE REVIEW

File: nemo/collections/common/data/lhotse/dataloader.py [hunk 1]
```diff
@@ -114,6 +115,11 @@ class LhotseDataLoadingConfig:
     cut_into_windows_hop: Optional[float] = None
     #       III) common options
     keep_excessive_supervisions: bool = True  # when a cut is truncated in the middle of a supervision, should we keep them.
+    #   e. RIR augmentation (synthetic RIR if rir_path is None)
+    #   at the moment supports only Lhotse recording manifests, e.g. https://github.com/lhotse-speech/lhotse/blob/master/lhotse/recipes/rir_noise.py
```
Comment (KunalDhawan) [inline nemo/collections/common/data/lhotse/dataloader.py:119]: Just opening a discussion thread here, would be great to support other alternatives in the future

**Review by KunalDhawan — APPROVED** [review]
Thank, LGTM!

File: nemo/collections/common/data/lhotse/dataloader.py (hunk 1 above)
Comment (pzelasko) [reply nemo/collections/common/data/lhotse/dataloader.py:119]: Yes, I'll aim to follow up shortly on this. I'll need to release a new lhotse version that extends the API to support CutSet RIR inputs first, and didn't want to block this work for that.

FWIW constructing a recording set is a one-liner `from lhotse import RecordingSet; RecordingSet.from_dir("my-rirs", pattern="*.wav").to_file("rirs.jsonl.gz")`
